packages/flutter_tools/lib/src/build_system/targets/android.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../artifacts.dart';
+import '../../base/file_system.dart';
+import '../../build_info.dart';
+import '../../globals.dart';
+import '../build_system.dart';
+import '../depfile.dart';
+import '../exceptions.dart';
+import 'assets.dart';
+import 'dart.dart';
+
+/// Prepares the asset bundle in the format expected by flutter.gradle.
+///
+/// The vm_snapshot_data, isolate_snapshot_data, and kernel_blob.bin are
+/// expected to be in the root output directory.
+///
+/// All assets and manifests are included from flutter_assets/**.
+abstract class AndroidAssetBundle extends Target {
+  const AndroidAssetBundle();
+
+  @override
+  List<Source> get inputs => const <Source>[
+    Source.pattern('{BUILD_DIR}/app.dill'),
+    Source.depfile('flutter_assets.d'),
+  ];
+
+  @override
+  List<Source> get outputs => const <Source>[
+    Source.depfile('flutter_assets.d'),
```

Comment:
Sorry if we discussed this already, but as a point of reference, GN targets have a 'depfile' parameter instead of requiring the depfile to be duplicated across inputs and outputs.

Said differently, what's the failure mode if someone forgets to include the depfile entry in both inputs and outputs?

Reply:
The current failure mode is that only the inputs or outputs of the depfile get added to the target. The fix would be to add a specific field to specify depfiles, which I can do as a part of this PR or on its own.

Reply:
Different PR sgtm